docs/src/components/table-of-contents.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  colorsBackgroundUtility,
+  colorsBackgroundLight
+} from '@pluralsight/ps-design-system-core'
+import { Heading } from '@pluralsight/ps-design-system-text'
+import Theme, { useTheme } from '@pluralsight/ps-design-system-theme'
+import VerticalTabs from '@pluralsight/ps-design-system-verticaltabs'
+import React from 'react'
+import GithubSlugger from 'github-slugger'
+
+import * as styles from './table-of-contents.module.css'
+
+export interface Heading {
+  value: string
+  depth: number
+}
+
+interface TableOfContentsProps {
+  headings: Heading[]
+}
+export const TableOfContents: React.FC<TableOfContentsProps> = props => {
+  const themeName = useTheme()
+  let slugger = new GithubSlugger()
+  React.useEffect(() => {
+    slugger = new GithubSlugger()
+  }, [slugger])
+  return (
+    <nav
+      className={styles.toc}
+      style={
+        {
+          '--bg':
+            themeName === Theme.names.dark
+              ? colorsBackgroundUtility[25]
+              : colorsBackgroundLight[2]
+        } as React.CSSProperties
```

Comment:
is this a new pattern we should follow. using css vars instead of toggling classes

Reply:
Oh, I'm not sure.  I was just geeking out about it.  Had seen it in the wild but never in a project like ours until types/prop.tsx was updated.

I think I still prefer class toggling.  Keeps the core out of the code and the style info in a stylesheet.

Do you think we should consolidate approaches?

Reply:
personally i'm not to worried about consolidating. let's see how it goes